app/controllers/api/providers_controller.rb

```diff
@@ -51,13 +51,11 @@ def refresh_resource(type, id = nil, _data = nil)
     end
 
     def delete_resource(type, id = nil, _data = nil)
-      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
-
-      api_action(type, id) do |klass|
-        provider = resource_search(id, type, klass)
-        api_log_info("Deleting #{provider_ident(provider)}")
-
-        destroy_provider(provider)
+      delete_action_handler do
+        raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+        provider = resource_search(id, type, collection_class(type))
+        task = provider.destroy_queue
+        action_result(true, "#{provider_ident(provider)} deleting", :task_id => task.id, :parent_id => id)
```

Comment:
why the parent_id here ?  that's different behavior the old destroy_provider method.

Reply:
@abellotti yeah, I'm confused as to how it was returning the `href`, because unless the `parent_id` is specified, it wasn't.